### PR TITLE
Sensor fault detection

### PR DESCRIPTION
**Pull request by daemonX10**
# Pull Request for ML-Crate 💡

## Issue Title: End-to-end Sensor Fault Detection #644

- **Info about the related issue (Aim of the project)** : The Sensor Fault Detection system is designed to monitor sensors and detect any faults. It uses advanced algorithms to ensure the accuracy and reliability of sensor data.
- **Name:** Damodar Yadav
- **Email ID for further communication**:  damodarryadav@gmail.com
- **GitHub ID:** https://github.com/daemonX10
- **Idenitfy yourself:** SSOC'24 contributor

Closes: #644

### Describe the add-ons or changes you've made 📃\

- Make the Model for Predication of condication of Sensor , it's good/Bad  need replacement or not . I used various model for this and xgboost model out perform other and then done hyperparameter tunning.
- Create PipeLine for data Ingestion , Data transformation , Model evaluation , Predication 
- Created Web app using Flask , django and Tailwind css  and more 

## What sort of change have you made:

- [   ] Bug fix (non-breaking change which fixes an issue)
- [✓] New feature (non-breaking change which adds functionality)
- [   ] Code style update (formatting, local variables)
- [   ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [   ] This change requires a documentation update

## How Has This Been Tested? ⚙️

_Describe how it has been tested_

_Describe how have you verified the changes made_

## Checklist: ☑️

- [✓] My code follows the guidelines of this project.
- [✓] I have performed a self-review of my own code.
- [✓] I have commented my code, particularly wherever it was hard to understand.
- [✓] I have made corresponding changes to the documentation.
- [✓] My changes generate no new warnings.
- [✓] I have added things that prove my fix is effective or that my feature works.
- [✓] Any dependent changes have been merged and published in downstream modules.

### CODE REVIEW

**Review by abhisheks008 — CHANGES_REQUESTED** [review]
Please follow the project structure and make the changes accordingly.
@daemonX10

**Comment by daemonX10** [comment]
@abhisheks008 changes done . i have make new folder for web part and remaining dataset , model and notebook along with image in main folder

**Comment by abhisheks008** [comment]
In the web app folder create a README file (template: https://github.com/abhisheks008/ML-Crate/blob/main/.github/web-app-readme-template.md) and put a demonstration video of the working of the web app. Also attach the same video in the README so that others can view that easily.

**Comment by daemonX10** [comment]
@abhisheks008  done those change in web readme file

**Comment by abhisheks008** [comment]
Just copy the demo video from your local system and paste the same inside the README so that it can be embedded inside the README and can also be viewed easily.
@daemonX10

**Comment by daemonX10** [comment]
that video is available in img folder already @abhisheks008

**Comment by abhisheks008** [comment]
Although it is available in the img folder, we can not view that video directly. That's why I told you to do so in the above comment.

**Comment by daemonX10** [comment]
@abhisheks008  added direct video embedding in readme

**Comment by abhisheks008** [comment]
image

Video is not visible. @daemonX10

**Comment by Harshit-code-tech** [comment]
i have 2 things to ask...
1) this is for local right??? how about web service deployment....it would be nice if you do that after submitting and completing on this...ofc on render or.. vercel.. 

2) i would like to know theway of it's working...it must be really frustating to make.. because of tons of errors...right?

**Comment by abhisheks008** [comment]
> i have 4 things to ask...
> 
> 1. this is for local right??? how about web service deployment....it would be nice if you do that after submitting and completing on this...ofc on render or.. vercel..
> 2. i would like to know theway of it's working...it must be really frustating to make.. because of tons of errors...right?
> 3. what are it's limitations.. ?
> 4. what's the best accuracy with val_loss?

@daemonX10 will explain this very well, as he is the one who is developing the whole thing. About the web app part, it is hosted in local. As a beginner when you are building a college project, locally hosted web app is enough, because majority of the college students used to take help from this project repository.

**Comment by daemonX10** [comment]
> image
> 
> Video is not visible. @daemonX10

**Comment by daemonX10** [comment]
> i have 4 things to ask...
> 
> 1. this is for local right??? how about web service deployment....it would be nice if you do that after submitting and completing on this...ofc on render or.. vercel..
> 2. i would like to know theway of it's working...it must be really frustating to make.. because of tons of errors...right?
> 3. what are it's limitations.. ?
> 4. what's the best accuracy with val_loss?

Local deployed, i have deployed multiple projects on render, you could ask me, although i have added all steps in Readme and the reason i don't deployed it, cause i not that much complex project.
2. NO, it's a simple project nothing more
3. Limitation, train on small dataset,
4. Val_loss check notebook

**Comment by daemonX10** [comment]
IMG_20240707_194929
@abhisheks008  see, i have added video

**Comment by Harshit-code-tech** [comment]
@daemonX10  to upload a video.. first go to any code issue tab.. in there from your local upload the video in chatbox...then github will provide a link..paste that in your readme note...

**Comment by abhisheks008** [comment]
> @daemonX10 to upload a video.. first go to any code issue tab.. in there from your local upload the video in chatbox...then github will provide a link..paste that in your readme note...
> 
> to check if your work or not.. go to file changed... in that .. go the the desired file... and click on 3 dots in right cornor of that particular...then view file...and then you can see what is visible to public
> 
> also... when used xgboost well..there is no issue in using decision tree..but it's no use when using xgboost...as it is the upgraded version of that

Upload the video as mentioned @daemonX10

**Comment by daemonX10** [comment]
https://github.com/abhisheks008/ML-Crate/assets/115637298/7712a38e-94b3-4463-ba65-c34f3e3a484d

updated this src link

**Comment by abhisheks008** [comment]
image

Still the video is not visible.

**Comment by daemonX10** [comment]
Do check now @abhisheks008, it showing

**Review by abhisheks008 — APPROVED** [review]
Hi @daemonX10 looks good to me. Approved ✅

**Comment by abhisheks008** [comment]
Hi @daemonX10 have you registered in the OSLead portal?

**Comment by daemonX10** [comment]
What is OSLead? @abhisheks008

**Comment by abhisheks008** [comment]
> What is OSLead? @abhisheks008

You are part of SSOC and you don't know OSLead? Check out the announcement channel of the SSOC Discord server and register yourself in OSLead for leader board.

**Comment by abhisheks008** [comment]
> > What is OSLead? @abhisheks008
> 
> You are part of SSOC and you don't know OSLead? Check out the announcement channel of the SSOC Discord server and register yourself in OSLead for leader board.
> 
> @daemonX10

Hi @daemonX10 have you done the needful?

**Comment by daemonX10** [comment]
I'm forgot that, can you give me link please

**Comment by daemonX10** [comment]
@abhisheks008
